### PR TITLE
sinon: add calledOnceWithMatch assertion

### PR DESCRIPTION
**Pull request by 43081j**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

---

This isn't documented yet, looks like they forgot to update their docs again when releasing 9.1.0

Basically they've introduced `calledOnceWithMatch` which is the lazy man's equivalent of:

```ts
assert.calledOnce(spy);
assert.calledWithMatch(spy, sinon.match.string);

// can now be
assert.calledOnceWithMatch(spy, sinon.match.string);
```

all these shortcuts can only end in clutter... 🙈

### CODE REVIEW

**Review by nicojs — APPROVED** [review]
Yeah. Now wait until they add `calledOnceWithAndCalledTriceWithAndNotCalledWithExceptOnAFullMoon`

**Comment by 43081j** [comment]
But.. but.. its less code right 😂 

Ready to merge.

**Comment by 43081j** [comment]
Ready to merge
